lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php

```diff
@@ -104,11 +106,32 @@ public function getDefinitions()
      * @param array $args
      *
      * @return object
-     *
+     * @throws \Exception
      */
     protected function createObject($type, $args)
     {
-        return new $type(...array_values($args));
+        try {
+            return new $type(...array_values($args));
+        } catch (\Throwable $e) {
+            $this->getLogger()->critical(__(
```

Comment:
It may be better, to replace `getLogger` method with a static object manager call, getting `LoggerInterface` right here, for example:
```php
use Magento\Framework\App\ObjectManager;
use Psr\Log\LoggerInterface;

$logger = ObjectManager::getInstance()->get(LoggerInterface::class);
```
This way it's possible not to rely on `$this->objectManager` which may or may not be set, making the code slightly more future-proof.